runelite-client/src/main/java/net/runelite/client/plugins/loginscreen/LoginScreenPlugin.java

```diff
@@ -245,59 +251,58 @@ private void overrideLoginScreen()
 	{
 		client.setShouldRenderLoginScreenFire(config.showLoginFire());
 
-		if (config.loginScreen() == LoginScreenOverride.OFF)
-		{
-			restoreLoginScreen();
-			return;
-		}
-
 		SpritePixels pixels = null;
-		if (config.loginScreen() == LoginScreenOverride.CUSTOM)
+		Random random = new Random(System.nanoTime());
+
+		switch (config.loginScreen())
 		{
-			if (CUSTOM_LOGIN_SCREEN_FILE.exists())
-			{
-				try
+			case OFF:
+				restoreLoginScreen();
+				return;
+			case CUSTOM:
+				if (CUSTOM_LOGIN_SCREEN_FILE.exists())
 				{
-					BufferedImage image;
-					synchronized (ImageIO.class)
-					{
-						image = ImageIO.read(CUSTOM_LOGIN_SCREEN_FILE);
-					}
+					pixels = getFileSpritePixels(CUSTOM_LOGIN_SCREEN_FILE);
+				}
+				break;
+			case RANDOM:
```

Comment:
For the RANDOM and RANDOM_CUSTOM options, it might be nice to choose a random image each time the login screen is reached, not just when the config is set. Not a must-have, but I feel that would be what users using this option would want.

Reply:
I just attempted to add this using the following code

```
if (event.getGameState() == GameState.LOGIN_SCREEN)
{
		applyUsername();

		if (config.loginScreen() == LoginScreenOverride.RANDOM ||
		config.loginScreen() == LoginScreenOverride.RANDOM_CUSTOM)
		{
				overrideLoginScreen();
		}
}
```
This results in an infinite loop as overrideLoginScreen(); causes a GameStateChanged.LOGIN_SCREEN event. I will have to think of an alternative method, for this functionality.

Reply:
You would have to set a boolean that gets reset when you log in.

Reply:
You would have to set a boolean that gets reset when you log in.

Reply:
This functionality has been added on latest commit